nautobot/core/apps/__init__.py

```diff
@@ -94,43 +106,65 @@ def register_menu_items(tab_list):
                         f"{nav_tab.name} -> {group.name} -> {item.link}",
```

Comment:
It's a bit awkward data-structure wise that we index the leaf menu items in the registry by their `link` rather than their `name` when the rest of the structure is indexed by `name`. Might consider changing this as it may allow for some code consolidation.